### ISSUE TITLE
aci_aaep_to_domain doesn't create relation between aaep and domain since version 2.6.0

### ISSUE THREAD

**Issue by Mike2912** [CiscoDevNet/terraform-provider-aci#1047]
**Terraform Version 1.4.6 
APIC version and APIC Platform 5.2(6e) on-prem
Provider-Version: 2.6.0, 2.6.1, 2.7.0
Affected Resource(s): aci_aaep_to_domain and aci_epg_to_contract** 

Hi all,

since we updated our aci provider from 2.5.2 to 2.7.0 the relation between aaep and domain with the ressource aci_aaep_to_domain does not work any more.
In provider version 2.6.0 it is flapping, so one time it is creating the relation and in the next run it's removing.
In provider version 2.7.0 the error changed to "The object may not exists"

Output:
Error: Error retrieving Object: Object may not exists
│ 
│   with module.portkonfig.aci_aaep_to_domain.map["1078"],
│   on ../modules/portkonfig/aaeps.tf line 19, in resource "aci_aaep_to_domain" "map":
│   19: resource "aci_aaep_to_domain" "map" {

The same issue was reported [here](https://discuss.hashicorp.com/t/terrafrom-aci-automation/52545)

The same happens to the relations between epg and contract from the ressource aci_epg_to_contract:
│ Error: Error retrieving Object: Object may not exists
│ 
│   with module.tenants.aci_epg_to_contract.map["1007"],
│   on ../modules/tenant/contract.tf line 70, in resource "aci_epg_to_contract" "map":
│   70: resource "aci_epg_to_contract" "map" {

Thanks in advance.

Regards 
Mike

**Comment by Felixgud** [comment]
Hi, 

after talking to an Cisco Engineer i post this update. Since there is now a version 2.8.0 all our tests belong to this new version. The last working release for our setup is still 2.5.2. With the engineer we removed the whole fabric config and deployed it from scratch with version 2.8.0. The error is the same as during the upgrade (see below).

Regarding the aci_aaep_to_domain resource. Unlike described above, in version 2.8.0 the connection between aaep and domain seems to no longer be removed. That seems to be addressed in #1045. Our problem number 2 seems to be the same just for other resources. 

Overall we have probably 2 problems with version 2.8.0.
1. A new problem with contracts makes terraform apply fail. The terraform code creates the contract resources in the following order:
- aci_contract
- aci_contract_subject
- aci_filter
- aci_contract_subject_filter
- aci_epg_to_contract
Before the deployment every ressource is already deployed and the config/terraform-code doesnt change. With the provider upgrade, terraform wants to deploy the resource aci_contract_subject again.
```
  # module.tenants.aci_contract_subject.map["Tenant-Permit"] will be created
  + resource "aci_contract_subject" "map" {
      + annotation                    = "orchestrator:terraform"
      + apply_both_directions         = "yes"
      + cons_match_t                  = "AtleastOne"
      + contract_dn                   = "uni/tn-Tenant/brc-EXT_CTR"
      + description                   = "created with Terraform"
      + id                            = (known after apply)
      + name                          = "Tenant_Permit"
      + name_alias                    = (known after apply)
      + prio                          = "unspecified"
      + prov_match_t                  = "AtleastOne"
      + relation_vz_rs_sdwan_pol      = (known after apply)
      + relation_vz_rs_subj_graph_att = (known after apply)
      + rev_flt_ports                 = "yes"
      + target_dscp                   = "unspecified"
    }
```
Because of the re-deployment of the aci_contract_subject the corresponding child object has also to be replaced.
```
# module.tenants.aci_contract_subject_filter.map["1000"] must be replaced
-/+ resource "aci_contract_subject_filter" "map" {
      + annotation          = "orchestrator:terraform"
      ~ contract_subject_dn = "uni/tn-Tenant/brc-EXT_CTR/subj-Tenant_Permit" # forces replacement -> (known after apply) # forces replacement
      + description         = (known after apply)
      ~ directives          = [
          - "",
          + "none",
        ]
      ~ id                  = "uni/tn-Tenant/brc-EXT_CTR/subj-Tenant_Permit/rssubjFiltAtt-Tenant_default" -> (known after apply)
        # (3 unchanged attributes hidden)
    }
```
This replacement may be the problem of the failed terraform apply. Terraform apply gives the following error message:
```
│ Error: Provider produced inconsistent result after apply
│ 
│ When applying changes to
│ module.tenants.aci_contract_subject.map["Tenant_Permit"], provider
│ "provider[\"registry.terraform.io/ciscodevnet/aci\"]" produced an
│ unexpected new value: Root resource was present, but now absent.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
```
On the web-ui you can see the connected subject to the contract. 


2. By doing a terraform plan, terraform wants to remove the - already existing - connection between
aci_external_network_instance_profile and aci_contract.
```
  # module.tenants.aci_external_network_instance_profile.map["1004"] will be updated in-place
  ~ resource "aci_external_network_instance_profile" "map" {
        id                           = "uni/tn-Tenant/out-L3Out_L3Out/instP-L3Out_EPG"
        name                         = "L3Out_EPG"
      ~ relation_fv_rs_prov          = [
          - "uni/tn-Tenant/brc-EXT_CTR",
        ]
        # (11 unchanged attributes hidden)
    }
```
The terraform code is written to create first the aci_external_network_instance_profile and aci_contract. Afterwards the aci_epg_to_contract resource creates the connection between the contract and the L3_Out-EPG.

**Comment by shrsr** [comment]
@Mike2912 @Felixgud Thank you for reporting the issue. We'll look into it and get this resolved as soon as we can.

**Comment by shrsr** [comment]
@Mike2912 Can you please try applying your configuration again with the latest 2.8.0 release and see if the issue still persists?

**Comment by shrsr** [comment]
@Felixgud Thanks for letting me know. I opened a new issue for the newer problems so that it's easier for us to track (#1055)
I'm closing this issue because the original problem is resolved as part of the fix #1045
